File: Client/karton.js
export class Karton {
    constructor(lbo, ime, prezime, id, datumIzdavanja) {
        this.ime = ime != null ? ime : "N";
        this.prezime = prezime != null ? prezime : "N";
        this.lbo = lbo != null ? lbo : "NN";
        this.id = id;
        this.datumIzdavanja = datumIzdavanja != null ? datumIzdavanja : new Date();
        this.terapije = [];
        this.container = null;

    }

    dodajTerapiju(terapija) {
        this.terapije.push(terapija);
    }

    crtajTerapije(roditeljskiDiv) {
        this.terapije.forEach((terapija) => {
            terapija.crtaj(roditeljskiDiv);
        })
    }

}

File: Models/Karton.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace Models {
        public class KartonPacijenta {
        [Key]
        public int Id { get; set; }

        [Required]
        [StringLength(12, ErrorMessage = "LBO must contain 12 characters.")]
        public string LBO { get; set; }
        
        [Required]
        [StringLength(60, MinimumLength = 3)]
        public string Ime { get; set; }

        [Required]
        [StringLength(60, MinimumLength = 3)]
        public string Prezime { get; set; }   

        [Required]
        public DateTime DatumIzdavanja { get; set; }

        public List<Terapija> Terapije {get; set;}

        [JsonIgnore]
        public KovidAmbulanta Ambulanta {get; set;}
    }
}